pyfr/integrators/dual/phys/controllers.py

```diff
@@ -47,8 +55,23 @@ def advance_to(self, t):
             raise ValueError('Advance time is in the past')
 
         while self.tcurr < t:
+            if self.tcurr + self._dt <= t < self.tcurr + self.i*self._dt:
```

Comment:
@semi-h Can you inspect this algorithm?

Reply:
We had a chat with @sambitmishra98 on how to best manage the changes in dt to meet the target t.

I think there are two important things
- We should aim to minimize the number of adjustments for the dt.
- And we should ideally stay as close as to the original dt as possible.

My idea is that we take an input from the ini file for the minimum factor we multiply dt to meet the target t. By default we can set it to 0.9 for example. Then, we'll need to monitor how far we are from the target and in case of factor 0.9 we should look to adjust dt when we're roughly 10 steps away from the target. What do you think about this @FreddieWitherden?

And on top of this Sambit mentioned that he wants to solve the machine precision related errors in t as well. So maybe we adjust the dt as I described above to roughly hit the target t, and then at the very last step we adjust dt one more time to get to the target t as accurately as possible.

I think the current implementation takes the input from ini file to determine how many time steps before we should adjust the dt. It'll be better if the input is the minimum factor that we multiply the dt, and then we figure out how many time steps before we need to take action.

Reply:
That seems sensible. Edge cases will need to be handled (for example when someone wants a plugin called so frequently that we do not have 10 steps to wok with).  Further, the code should be shared between std and dual integrators.